### PR TITLE
Use thriftrw for tcollector.

### PR DESCRIPTION
**Pull request by abhinav**
Required for #199.

CC @breerly @blampe @junchaowu

### CODE REVIEW

**Comment by junchaowu** [comment]
nice lgtm

**Comment by breerly** [comment]
fucking yes :clap:

**Comment by blampe** [comment]
:+1:

File: tchannel/zipkin/tcollector.thrift [hunk 1]
```diff
@@ -37,28 +37,28 @@ enum AnnotationType { BOOL, BYTES, I16, I32, I64, DOUBLE, STRING }
...
-    6: list<Annotation> annotations
-    7: list<BinaryAnnotation> binaryAnnotations
-    8: optional bool debug = 0
+    1: required binary traceId           // unique trace id, use for all spans in trace
+    2: required Endpoint host            // host being remotely procedure called
+    3: required string name              // span name, rpc method for example
+    4: required binary id                // unique span id, only used for this span
+    5: required binary parentId          // parent span id, 0 if no parent
+    6: required list<Annotation> annotations
+    7: required list<BinaryAnnotation> binaryAnnotations
```
Comment (breerly) [inline tchannel/zipkin/tcollector.thrift]: are all these actually required?

**Pull request merged**